app/com/arpnetworking/rollups/RollupManager.java

```diff
@@ -77,6 +85,41 @@ public Receive createReceive() {
                 .build();
     }
 
+    private void executorFinished(final RollupExecutor.FinishRollupMessage message) {
+        final Optional<Throwable> failure = message.getFailure();
+        _periodicMetrics.recordCounter("rollup/manager/executor_finished/success", failure.isPresent() ? 0 : 1);
+        if (!failure.isPresent()) {
+            return;
+        }
+
+        final boolean isRetryable = RollupPartitioningUtils.mightSplittingFixFailure(failure.get());
+        _periodicMetrics.recordCounter("rollup/manager/executor_finished/unretriable", isRetryable ? 0 : 1);
+        if (!isRetryable) {
+            LOGGER.warn()
+                    .setMessage("giving up after non-retryable error")
+                    .addData("rollupDefinition", message.getRollupDefinition())
+                    .setThrowable(failure.get())
+                    .log();
+            return;
+        }
+
```

Comment:
Can we at a minimum log which metrics we're retrying/splitting.  Ideally we'd have a metric by metricname but if we're concerned about cardinality then I'm okay with just logging.  The idea is that this may help inform us on where we would want to look if we switched to pre-splitting via configuration or some other mechanism.